### ISSUE TITLE
HTTParty shows "head and body had been removed intentionally"

### ISSUE THREAD

**Issue by LucasCioffi** [ether/etherpad-lite#2667]
In the console, when I use HTTParty to grab the HTML from a pad like this...

    HTTParty.get("url_to_the_pad")

I get the HTML as a result but I see "head and body had been removed intentionally" inside the results.  This is the first 10% of the HTML for example:

    "\n<!doctype html>\n\n<html>\n\n\n        <title>Notes</title>\n        <script>\n          /*\n          |@licstart  The following is the entire license notice for the\n          JavaScript code in this page.|\n\n          Copyright 2011 Peter Martischka, Primary Technology.\n\n          Licensed under the Apache License, Version 2.0 (the \"License\");\n          you may not use this file except in compliance with the License.\n          You may obtain a copy of the License at\n\n             http://www.apache.org/licenses/LICENSE-2.0\n\n          Unless required by applicable law or agreed to in writing, software\n          distributed under the License is distributed on an \"AS IS\" BASIS,\n          WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n          See the License for the specific language governing permissions and\n          limitations under the License.\n\n          |@licend  The above is the entire license notice\n          for the JavaScript code in this page.|\n          */\n        </script>\n\n        <meta charset=\"utf-8\">\n        <meta name=\"robots\" content=\"noindex, nofollow\">\n        <meta name=\"viewport\" content=\"width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0\">\n        <link rel=\"shortcut icon\" href=\"../favicon.ico\">\n\n        \n        <link href=\"../static/css/pad.css\" rel=\"stylesheet\">\n\n        \n        <link href=\"../static/custom/pad.css\" rel=\"stylesheet\">\n        \n\n        <style title=\"dynamicsyntax\"></style>\n        \n\n        <link rel=\"localizations\" type=\"application/l10n+json\" href=\"../locales.json\" />\n        <script type=\"text/javascript\" src=\"../static/js/html10n.js\"></script>\n        <script type=\"text/javascript\" src=\"../static/js/l10n.js\"></script>\n\n        \n\n        \n        <div id=\"editbar\" class=\"toolbar\" title=\"Toolbar (Alt F9)\">\n            <div id=\"overlay\">\n                <div id=\"overlay-inner\"></div>\n            </div>\n\n            <ul class=\"menu_left\" role=\"toolbar\">\n                \n                <li data-type=\"button\" data-key=\"bold\"><a class=\"grouped-left\" data-l10n-id=\"pad.toolbar.bold.title\"><button class=\" buttonicon buttonicon-bold\" data-l10n-id=\"pad.toolbar.bold.title\"></button></a></li>\n<li data-type=\"button\" data-key=\"italic\"><a class=\"grouped-middle\" data-l10n-id=\"pad.toolbar.italic.title\"><button class=\" buttonicon buttonicon-italic\" data-l10n-id=\"pad.toolbar.italic.title\"></button></a></li>\n<li data-type=\"button\" data-key=\"underline\"><a class=\"grouped-middle\" data-l10n-id=\"pad.toolbar.underline.title\"><button class=\" buttonicon buttonicon-underline\" data-l10n-id=\"pad.toolbar.underline.title\"></button></a></li>\n<li data-type=\"button\" data-key=\"strikethrough\"><a class=\"grouped-right\" data-l10n-id=\"pad.toolbar.strikethrough.title\"><button class=\" buttonicon buttonicon-strikethrough\" data-l10n-id=\"pad.toolbar.strikethrough.title\"></button></a></li><li class=\"separator\"></li><li data-type=\"button\" data-key=\"insertorderedlist\"><a class=\"grouped-left\" data-l10n-id=\"pad.toolbar.ol.title\"><button class=\" buttonicon buttonicon-insertorderedlist\" data-l10n-id=\"pad.toolbar.ol.title\"></button></a></li>\n<li data-type=\"

Is that message "head and body had been removed intentionally" coming from EtherPad or HTTParty?  I didn't see that phrase in the source of this etherpad-lite repository.

**Comment by JohnMcLear** [comment]
I see body in export.  http://beta.etherpad.org/p/test/export/html

**Comment by JohnMcLear** [comment]
Dunno what you are doing to make it not appear..  Assuming it's a HTTParty bug/feature..

**Comment by LucasCioffi** [comment]
John, thanks for the prompt response.  I found a work around.  Here's what I found in case it's helpful.  The reason I'm using HTTParty is to get the HTML for a pad so that I can parse it with Nokogiri and email the pad contents to a user from time to time.

Entering the following command in to the console was still hiding the head and body in the output with the phrase "head and body had been removed intentionally":

    HTTParty.get("https://beta.etherpad.org/p/test")

It seems that the phrase "head and body had been removed intentionally" is coming from Etherpad, because Google search results on that phrase only turn up websites related to Etherpad.

When I changed the URL to be the HTML export URL, I was able to successfully see the header and body in the returned results:

    HTTParty.get("https://beta.etherpad.org/p/test/export/html")

I never saw the phrase "head and body had been removed intentionally" when I tried this on Mozilla's non-lite (I think) version of Etherpad:

    HTTParty.get("https://etherpad.mozilla.org/GzI9USKR6W")

So there's no issue here, but this might be helpful for others trying to screen scrape their pads.  Thank you!
